MIG-1687: Release notes for MTC 1.8.6 (#91067)
openshift/openshift-docs: modules/migration-mtc-release-notes-1-8-6.adoc

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/mtc-release-notes.adoc
:_mod-docs-content-type: REFERENCE
[id="migration-mtc-release-notes-1-8-6_{context}"]
= {mtc-full} 1.8.4 release notes

[id="new-features-1-8-6_{context}"]
== New features

> 

Author: ```suggestion
.Multiple migration plans for a namespace

Before the release of {mtc-short} 1.8.6, you were unable to have more than one migration plan for a namespace.  If another migration plan attempted to use a namespace that was associated with another migration plan, the plan would be rejected. 

In {mtc-short} 1.8.6, you can have multiple migration plans associated with the same namespace, provided that your migration plans do not reference the same persistent volumes (PVs). 

By having multiple migration plans associated with the same namespace, you can design different migration plans for applications that reside in the same namespace.
```

.Unable to create a migration plan due to a reconciliation failure

When creating a migration plan in {mtc-short}, the UI becomes stuck on the *Persistent Volumes* and you cannot continue. This issue occurs due to a critical reconciliation failure and returns a 404 API error when you attempt to fetch the migration plan from the backend. These issues cause the migration plan to remain in a *Not Ready* state, and you are prevented from continuing. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1705[(MIG-1705)]

.Migration process becomes stuck after the `StageBackup` phase

When migrating a Django and PostgreSQL application, the migration becomes stuck after the `StageBackup` phase. Even though all the pods in the source namespace are healthy before the migration begins, after the migration and when terminating the pods on the source cluster, the Django pod fails with a `CrashLoopBackOff` error. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1707[(MIG-1707)]

> 

Author: ```suggestion
.Migration shown as succeeded despite a failed phase due to a misleading UI status

After running a migration using {mtc-short}, the UI incorrectly indicates that the migration was successful, with the status shown as *Migration succeeded*. However, the Direct Volume Migration (DVM) phase failed. This misleading status appears on both the *Migration* and the *Migration Details* pages. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1711[(MIG-1711)] 
```

.Migration process becomes stuck after the `StageBackup` phase

When migrating a Django and PostgreSQL application, the migration becomes stuck after the `StageBackup` phase. Even though all the pods in the source namespace are healthy before the migration begins, after the migration and when terminating the pods on the source cluster, the Django pod fails with a `CrashLoopBackOff` error. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1707[(MIG-1707)]

.Migration shown as succeeded despite a failed phase due to a misleading UI status

After running a migration using {mtc-short}, the UI incorrectly indicates that the migration was successful, with the status shown as *Migration succeeded*. However, the Direct Volume Migration (DVM) phase failed. This misleading status appears on both the *Migration* and the *Migration Details* pages. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1711[(MIG-1711)] 

[id="known-issues-1-8-6_{context}"]
== Known issues

{mtc-short} 1.8.6 has the following known issues:

.Inconsistent reporting of migration failure status

> There is a discrepancy in the reporting of namespace migration status following a rollback and subsequent re-migration attempt when the migration plan is deliberately faulted. Although the Distributed Volume Migration (DVM) phase correctly registers a failure, this failure is not consistently reflected in the user interface (UI) or the migration plan's YAML representation. link:https://issues.redhat.com/browse/MIG-1719[(MIG-1719)]

Reviewer: ```suggestion
There is a discrepancy in the reporting of namespace migration status following a rollback and subsequent re-migration attempt when the migration plan is deliberately faulted. Although the Distributed Volume Migration (DVM) phase correctly registers a failure, this failure is not consistently reflected in the user interface (UI) or the migration plan's YAML representation. link:https://issues.redhat.com/browse/MIG-1719[(MIG-1719)]

Additionally, this issue is not limited to edge cases. In certain scenarios—such as when network restrictions are applied that cause the DVM phase to fail—the UI still reports the migration status as succeeded. This behavior is similar to what was previously observed in MIG-1711 but occurs under different conditions, indicating that multiple root causes may lead to the same incorrect UI status. Both cases warrant documentation to clarify the limitations of status reporting in such failure scenarios.
```

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/mtc-release-notes.adoc
:_mod-docs-content-type: REFERENCE
[id="migration-mtc-release-notes-1-8-6_{context}"]
= {mtc-full} 1.8.6 release notes

[id="new-features-1-8-6_{context}"]
== New features

.Multiple migration plans for a namespace

Before the release of {mtc-short} 1.8.6, you were unable to have more than one migration plan for a namespace. If another migration plan attempted to use a namespace that was associated with another migration plan, the plan was rejected. 

In {mtc-short} 1.8.6, you can have more than one migration plans associated with the same namespace, if your migration plans do not reference the same persistent volumes (PVs). 

By having more than one migration plan associated with the same namespace, you can design different migration plans for applications that are in the same namespace.

[id="resolved-issues-1-8-6_{context}"]
== Resolved issues

{mtc-short} 1.8.6 has the following major resolved issues:

> .UI crashes during a namespace search in migration plan wizard

Author: ```suggestion
.UI crashes during a namespace search in the migration plan wizard
```

.Unable to create a migration plan due to a reconciliation failure

When creating a migration plan in {mtc-short}, the UI becomes stuck on the *Persistent Volumes* and you cannot continue. This issue occurs due to a critical reconciliation failure and returns a 404 API error when you attempt to fetch the migration plan from the backend. These issues cause the migration plan to remain in a *Not Ready* state, and you are prevented from continuing. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1705[(MIG-1705)]

.Migration process becomes stuck after the `StageBackup` phase

When migrating a Django and PostgreSQL application, the migration becomes stuck after the `StageBackup` phase. Even though all the pods in the source namespace are healthy before the migration begins, after the migration and when terminating the pods on the source cluster, the Django pod fails with a `CrashLoopBackOff` error. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1707[(MIG-1707)]

> .Migration shown as succeeded despite a failed phase due to a misleading UI status

Author: ```suggestion

.Migration shown as succeeded despite a failed phase due to a misleading UI status
```

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/mtc-release-notes.adoc
:_mod-docs-content-type: REFERENCE
[id="migration-mtc-release-notes-1-8-6_{context}"]
= {mtc-full} 1.8.6 release notes

[id="technical-changes-1-8-6_{context}"]
== Technical changes

.Multiple migration plans for a namespace is not supported

{mtc-short} version 1.8.6 and later do not support multiple migration plans for a single namespace.

.VM storage migration

> VM storage migration feature changes from Technical Preview status to being Generally Available.

Author: ```suggestion
VM storage migration feature changes from Technology Preview (TP) status to being Generally Available (GA).
```

> VM storage migration feature changes status from Technical Preview status to being generally available.

Reviewer 2: ```suggestion
VM storage migration feature changes from Technical Preview status to being Generally Available.
```
Should "Generally Available" here be capitalized similar to "Technical Preview"?
Remove one of the use of 'status' here?

.Migration process becomes stuck after the `StageBackup` phase

When migrating a Django and PostgreSQL application, the migration becomes stuck after the `StageBackup` phase. Even though all the pods in the source namespace are healthy before the migration begins, after the migration and when terminating the pods on the source cluster, the Django pod fails with a `CrashLoopBackOff` error. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1707[(MIG-1707)]

.Migration shown as succeeded despite a failed phase due to a misleading UI status

After running a migration using {mtc-short}, the UI incorrectly indicates that the migration was successful, with the status shown as *Migration succeeded*. However, the Direct Volume Migration (DVM) phase failed. This misleading status appears on both the *Migration* and the *Migration Details* pages. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1711[(MIG-1711)] 

[id="known-issues-1-8-6_{context}"]
== Known issues

{mtc-short} 1.8.6 has the following known issues:

> 

Reviewer: we need to document the [MIG-1719](https://issues.redhat.com/browse/MIG-1719) issue

> 

Reviewer: we need to document the [MIG-1719](https://issues.redhat.com/browse/MIG-1719) issue

// Module included in the following assemblies:
//
// * migration_toolkit_for_containers/mtc-release-notes.adoc
:_mod-docs-content-type: REFERENCE
[id="migration-mtc-release-notes-1-8-6_{context}"]
= {mtc-full} 1.8.6 release notes

[id="technical-changes-1-8-6_{context}"]
== Technical changes

.Multiple migration plans for a namespace is not supported

{mtc-short} version 1.8.6 and later do not support multiple migration plans for a single namespace.

.VM storage migration

VM storage migration feature changes from Technology Preview (TP) status to being Generally Available (GA).

[id="resolved-issues-1-8-6_{context}"]
== Resolved issues

{mtc-short} 1.8.6 has the following major resolved issues:

> .UI crashes during a namespace search in the migration plan wizard

Reviewer 3: Try to use a more descriptive term for "crash", such as "fails" or "ends unexpectedly". IBM SG.

> .UI crashes during a namespace search in the migration plan wizard

Author: ```suggestion
.UI fails during a namespace search in the migration plan wizard
```

> When searching for a namespace in the *Select Namespace* step of the migration plan wizard, the user interface (UI) crashes and disappears after clicking *Search*. The browser console shows a JavaScript error indicating that an undefined value has been accessed. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1704[(MIG-1704)]

Author: ```suggestion
When searching for a namespace in the *Select Namespace* step of the migration plan wizard, the user interface (UI) fails and disappears after clicking *Search*. The browser console shows a JavaScript error indicating that an undefined value has been accessed. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1704[(MIG-1704)]
```

> When searching for a namespace in the *Select Namespace* step of the migration plan wizard, the user interface (UI) crashes and disappears after clicking *Search*. The browser console shows a JavaScript error indicating that an undefined value has been accessed. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1704[(MIG-1704)]

Reviewer 3: Same comment as above.

.Unable to create a migration plan due to a reconciliation failure

> When creating a migration plan in {mtc-short}, the UI becomes stuck on the *Persistent Volumes* and you cannot continue. This issue occurs due to a critical reconciliation failure and returns a 404 API error when you attempt to fetch the migration plan from the backend. These issues cause the migration plan to remain in a *Not Ready* state, and you are prevented from continuing. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1705[(MIG-1705)]

Reviewer 3: ```suggestion
In {mtc-short}, when creating a migration plan , the UI remains on *Persistent Volumes* and you cannot continue. This issue occurs due to a critical reconciliation failure and returns a 404 API error when you attempt to fetch the migration plan from the backend. These issues cause the migration plan to remain in a *Not Ready* state, and you are prevented from continuing. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1705[(MIG-1705)]
```

> .Migration process becomes stuck after the `StageBackup` phase

Author: ```suggestion
.Migration process becomes fails to complete after the `StageBackup` phase
```

> When migrating a Django and PostgreSQL application, the migration becomes stuck after the `StageBackup` phase. Even though all the pods in the source namespace are healthy before the migration begins, after the migration and when terminating the pods on the source cluster, the Django pod fails with a `CrashLoopBackOff` error. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1707[(MIG-1707)]

Reviewer 3: Same comment as above.

> When migrating a Django and PostgreSQL application, the migration becomes stuck after the `StageBackup` phase. Even though all the pods in the source namespace are healthy before the migration begins, after the migration and when terminating the pods on the source cluster, the Django pod fails with a `CrashLoopBackOff` error. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1707[(MIG-1707)]

Author: ```suggestion
When migrating a Django and PostgreSQL application, the migration becomes fails to complete after the `StageBackup` phase. Even though all the pods in the source namespace are healthy before the migration begins, after the migration and when terminating the pods on the source cluster, the Django pod fails with a `CrashLoopBackOff` error. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1707[(MIG-1707)]
```

.Migration shown as succeeded despite a failed phase due to a misleading UI status

After running a migration using {mtc-short}, the UI incorrectly indicates that the migration was successful, with the status shown as *Migration succeeded*. However, the Direct Volume Migration (DVM) phase failed. This misleading status appears on both the *Migration* and the *Migration Details* pages. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1711[(MIG-1711)] 

> .Persistent Volumes page hangs indefinitely for namespaces without PVCs

Reviewer 3: ```suggestion
.Persistent Volumes page remains indefinitely for namespaces without PVCs
```
Same comment as above.

Reviewer 3: The acronym PVC is not spelled out here, but is below. Not sure what the convention is here. Do you spell out on first instance for the topic, or for each bug?

> .Persistent Volumes page hangs indefinitely for namespaces without PVCs

Author: ```suggestion
.Persistent Volumes page hangs indefinitely for namespaces without persistent volume claims
```

> When a migration plan includes a namespace that does not have any persistent volume claims (PVCs), the *Persistent Volumes* selection page becomes stuck with the following message shown: `Discovering persistent volumes attached to source projects...`. The page never completes loading, preventing you from proceeding with the migration. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1713[(MIG-1713)]

Reviewer 3: ```suggestion
When a migration plan includes a namespace that does not have any persistent volume claims (PVCs), the *Persistent Volumes* selection page remains indefinitely with the following message shown: `Discovering persistent volumes attached to source projects...`. The page never completes loading, preventing you from proceeding with the migration. This issue has been resolved in {mtc-short} 1.8.6. link:https://issues.redhat.com/browse/MIG-1713[(MIG-1713)]
```

[id="known-issues-1-8-6_{context}"]
== Known issues

{mtc-short} 1.8.6 has the following known issues:

.Inconsistent reporting of migration failure status

There is a discrepancy in the reporting of namespace migration status following a rollback and subsequent re-migration attempt when the migration plan is deliberately faulted. Although the Distributed Volume Migration (DVM) phase correctly registers a failure, this failure is not consistently reflected in the user interface (UI) or the migration plan's YAML representation. 

> This issue is not only limited to edge cases. In certain circumstances, such as when network restrictions are applied that cause the DVM phase to fail, the UI still reports the migration status as successful. This behavior is similar to what was previously observed in link:https://issues.redhat.com/browse/MIG-1711[MIG-1711] but occurs under different conditions. link:https://issues.redhat.com/browse/MIG-1719[(MIG-1719)] 

Reviewer 3: Not sure about using the term "edge case". Not everyone will know what that means and it will translate poorly. Perhaps "unusual cases"?

> This issue is not only limited to edge cases. In certain circumstances, such as when network restrictions are applied that cause the DVM phase to fail, the UI still reports the migration status as successful. This behavior is similar to what was previously observed in link:https://issues.redhat.com/browse/MIG-1711[MIG-1711] but occurs under different conditions. link:https://issues.redhat.com/browse/MIG-1719[(MIG-1719)] 

Author: ```suggestion
This issue is not only limited to  to unusual or unexpected cases. In certain circumstances, such as when network restrictions are applied that cause the DVM phase to fail, the UI still reports the migration status as successful. This behavior is similar to what was previously observed in link:https://issues.redhat.com/browse/MIG-1711[MIG-1711] but occurs under different conditions. link:https://issues.redhat.com/browse/MIG-1719[(MIG-1719)] 
```
